samples/nrf_desktop/src/modules/user_config.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <assert.h>
+
+#include <zephyr.h>
+
+#define MODULE user_config
+
+#include "config_event.h"
+#include "motion_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USER_CONFIG_LOG_LEVEL);
+
+enum {
+	USER_CONFIG_ID_RESERVED,
+	USER_CONFIG_ID_MOUSE_CPI
+};
+
+static struct user_cfg {
+	u16_t mouse_cpi; /* Mouse optical sensor resolution (counts per inch) */
+} user_cfg;
+
+static void user_cfg_process(u8_t id, u8_t *data)
+{
+	switch (id) {
+	case USER_CONFIG_ID_MOUSE_CPI:
```

Comment:
I am against doing it this way as we need to update this file every time new config option is added

what can be done:
1. you create a config event and send it as you do
2. any module that needs to react on config change hooks to config event and if such is received it checks if this config is addressed to it (you can keep the enum with config message ids in the config event header).
3. module processes the event as it wishes - this is done locally to where data is used

Reply:
Sounds like a good idea.